pkg/resourcemanager/azuresql/azuresqlmanageduser/azuresqlmanageduser_reconcile.go

```diff
@@ -81,9 +81,8 @@ func (s *AzureSqlManagedUserManager) Ensure(ctx context.Context, obj runtime.Obj
 			return false, nil
 		}
 
-		// Other failures are terminal
 		instance.Status.SetFailedProvisioning(instance.Status.Message)
-		return true, nil
+		return false, nil
```

Comment:
Ergh. I dislike returning boolean values unless they have really simple meanings - better to define an enumeration and be declarative. Something for later, perhaps.

Reply:
Agreed! Especially here for Ensure and for Delete - the meaning of these booleans are pretty unclear, you basically need to read the AsyncReconciler code to see how they're used. I'm going to make an issue to clean them up - but obviously it's spread across all of the manager types unfortunately.